vm_method.c

```diff
@@ -509,10 +500,21 @@ setup_method_cfunc_struct(rb_method_cfunc_t *cfunc, VALUE (*func)(ANYARGS), int
     cfunc->invoker = call_cfunc_invoker_func(argc);
 }
 
+static rb_method_definition_t *
+method_definition_addref(rb_method_definition_t *def, bool complemented)
+{
+    def->reference_count++;
+    if (!complemented)
+        def->aliased = true;
```

Comment:
I think this should be:

```suggestion
    if (!complemented &&  def->reference_count > 0) def->aliased = true;
    def->reference_count++;
```

We don't want to set `aliased` for the very first reference to `rb_method_definition_t` when it was just created.